src/github.com/matrix-org/dendrite/federationsender/consumers/roomserver.go

```diff
@@ -134,25 +134,27 @@ func (s *OutputRoomEvent) processMessage(ore api.OutputNewRoomEvent) error {
 		return err
 	}
 
+	if oldJoinedHosts == nil {
+		// This means that there is nothing to update as this is a duplicate
+		// message.
```

Comment:
I think it's worth commenting on why this situation arises. If it's only because dendrite crashed between reading the message and persisting the stream position, say so.

Reply:
(Done)